src/types/datetime.rs

```diff
@@ -524,4 +556,33 @@ mod tests {
             assert!(b.unwrap().get_fold());
         });
     }
+
+    #[cfg(not(PyPy))]
+    #[test]
+    fn test_get_tzinfo() {
+        crate::Python::with_gil(|py| {
+            use crate::conversion::ToPyObject;
+            use crate::types::{PyDateTime, PyTime, PyTzInfoAccess};
+
+            let datetime = py.import("datetime").map_err(|e| e.print(py)).unwrap();
+            let timezone = datetime.getattr("timezone").unwrap();
+            let utc = timezone.getattr("utc").unwrap().to_object(py);
+
+            let dt = PyDateTime::new(py, 2018, 1, 1, 0, 0, 0, 0, Some(&utc)).unwrap();
+
+            assert!(dt.get_tzinfo().unwrap().eq(&utc).unwrap());
+
+            let dt = PyDateTime::new(py, 2018, 1, 1, 0, 0, 0, 0, None).unwrap();
+
+            assert!(dt.get_tzinfo().is_none());
+
+            let t = PyTime::new(py, 0, 0, 0, 0, Some(&utc)).unwrap();
+
+            assert!(t.get_tzinfo().unwrap().eq(&utc).unwrap());
+
+            let t = PyTime::new(py, 0, 0, 0, 0, None).unwrap();
+
+            assert!(t.get_tzinfo().is_none());
```

Comment:
I have no idea why this is not none, need to investigate.

Reply:
I think I might need some help here, still kind confuse why this did not work.

```rust
crate::py_run!(t.py(), t, "assert t.tzinfo is None");
```

This did not error but when it got to rust side it errors. I further check and saw that `PyDateTime_TIME_GET_TZINFO(self.as_ptr())` returns `0x0000000000000001` even though it should be `None`, I wonder why it isn't `0x0000000000000000` which was returned by `PyDateTime_DATE_GET_TZINFO(self.as_ptr())`.

Reply:
@pickfire I've opened #2289 to resolve, I don't think we need to worry about the different `BaseTime` struct layout as long as we check `has_tzinfo` first.